tests/jballerina-unit-test/src/test/resources/test-src/javainterop/dependently_typed_functions_test.bal

```diff
@@ -891,6 +891,31 @@ function testDependentlyTypedFunctionWithInferredArgForParamOfTypeReferenceType(
     assert("hello!", d);
 }
 
+client class ClientWithExternalResourceBody {
+    resource function get [string... path](TargetType targetType) returns targetType|error =
+    @java:Method {
+        'class: "org.ballerinalang.nativeimpl.jvm.tests.VariableReturnType",
+        name: "getResource"
+    } external;
+}
+
+function testDependentlyTypedResourceFunctions() {
+    ClientWithExternalResourceBody cl = new ClientWithExternalResourceBody();
+    string|error a = cl->/games/carrom(targetType = string);
+    assert("[\"games\",\"carrom\"]", checkpanic a);
+    
+    var cl2 = client object {
+        resource function get [string... path](TargetType targetType) returns targetType|error =
+        @java:Method {
+                'class: "org.ballerinalang.nativeimpl.jvm.tests.VariableReturnType",
+                name: "getResource"
+            } external;
+    };
+
+    string|error b = cl2->/games/football(targetType = string);
```

Comment:
We need to test for inferred param with and without the argument also.

```ballerina
client class ClientWithExternalResourceBody2 {
    resource function get [string... path](TargetType targetType = <>) returns targetType|error =
    @java:Method {
        'class: "org.ballerinalang.nativeimpl.jvm.tests.VariableReturnType",
        name: "getResource"
    } external;
}

    // both should work
    string|error b = cl2->/games/football(targetType = string);
    // .... 
    int|error c = cl2->/games/football();
    // ....
```

Also in the Java method, I would assert the tag to make sure the typedesc arg changes accordingly.

Reply:
Added with [5697abb](https://github.com/ballerina-platform/ballerina-lang/pull/37042/commits/5697abbcb73f69f7407a7f7f1efbdf10ff1c9238)